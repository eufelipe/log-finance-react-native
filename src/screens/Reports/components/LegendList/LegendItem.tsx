import {Currency} from 'components';
import React from 'react';

import {ReportItem} from 'services/reports';

import {Container, Title, SubTitle, Circle} from './styles';

const LegendItem = ({text, value, sumed, color}: ReportItem): JSX.Element => {
  return (
    <Container>
      <Circle color={color} />

      <Title>
        {text} ({sumed})
      </Title>

      <Currency
        value={value}
        render={currency => <SubTitle>{currency}</SubTitle>}
      />
    </Container>
  );
};

export default LegendItem;
