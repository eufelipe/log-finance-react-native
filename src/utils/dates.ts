import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
