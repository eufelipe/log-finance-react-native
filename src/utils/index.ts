export {default as number} from './number';
export {default as strings} from './strings';
