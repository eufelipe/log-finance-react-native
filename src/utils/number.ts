export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
