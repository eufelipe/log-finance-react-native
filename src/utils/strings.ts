export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
