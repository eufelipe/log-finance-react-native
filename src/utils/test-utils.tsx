/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

import {NavigationContainer} from '@react-navigation/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
