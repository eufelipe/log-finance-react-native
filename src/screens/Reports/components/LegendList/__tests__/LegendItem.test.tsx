/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/unbound-method */
import React from 'react';

import {render} from 'utils/test-utils';
import LegendItem from '../LegendItem';

const MOCK = {
  id: '123abc',
  color: 'red',
  text: 'Teste',
  sumed: 100,
  value: 10,
};

describe('LegendItem', () => {
  it('Test match snapshot LegendItem', () => {
    // given
    const props = {
      ...MOCK,
    };

    // when
    const {toJSON} = render(<LegendItem {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
