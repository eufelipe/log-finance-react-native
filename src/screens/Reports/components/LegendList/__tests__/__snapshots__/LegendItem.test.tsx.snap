// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendItem Test match snapshot LegendItem 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 5,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 5,
        "width": "50%",
      },
    ]
  }
>
  <View
    color="red"
    style={
      Array [
        Object {
          "backgroundColor": "red",
          "borderRadius": 8,
          "height": 16,
          "marginRight": 5,
          "width": 16,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#808080",
          "fontFamily": "Poppins-Regular",
          "fontSize": 20,
          "fontWeight": "700",
          "marginRight": 5,
        },
      ]
    }
  >
    Teste
     (
    100
    )
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#808080",
          "fontFamily": "Poppins-Regular",
          "fontSize": 20,
          "fontWeight": "400",
          "marginRight": 5,
        },
      ]
    }
  >
    R$Â 0,10
  </Text>
</View>
`;
